lib/ethpy/ethpy/hyperdrive/api/api_test.py

```diff
@@ -23,102 +25,208 @@
         # TODO: remove cast when pypechain consolidates dataclasses.
-        pool_config = cast(PoolConfig, hyperdrive.hyperdrive_contract.functions.getPoolConfig().call())
-        assert pool_config_to_fixedpoint(pool_config) == hyperdrive.current_pool_state.pool_config
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        pool_config = cast(PoolConfig, hyperdrive_interface.hyperdrive_contract.functions.getPoolConfig().call())
+        assert pool_config_to_fixedpoint(pool_config) == hyperdrive_interface.current_pool_state.pool_config
 
     def test_pool_info(self, local_hyperdrive_pool: DeployedHyperdrivePool):
         """Checks that the Hyperdrive pool_info matches what is returned from the smart contract.
 
         All arguments are fixtures.
         """
-        uri: URI | None = cast(HTTPProvider, local_hyperdrive_pool.web3.provider).endpoint_uri
-        rpc_uri = uri if uri else URI("http://localhost:8545")
-        abi_dir = "./packages/hyperdrive/src/abis"
-        hyperdrive_contract_addresses: HyperdriveAddresses = local_hyperdrive_pool.hyperdrive_contract_addresses
-        eth_config = EthConfig(artifacts_uri="not used", rpc_uri=rpc_uri, abi_dir=abi_dir)
-        hyperdrive = HyperdriveInterface(eth_config, addresses=hyperdrive_contract_addresses)
-        pool_info = hyperdrive.hyperdrive_contract.functions.getPoolInfo().call()
-        assert pool_info_to_fixedpoint(pool_info) == hyperdrive.current_pool_state.pool_info
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        pool_info = hyperdrive_interface.hyperdrive_contract.functions.getPoolInfo().call()
+        assert pool_info_to_fixedpoint(pool_info) == hyperdrive_interface.current_pool_state.pool_info
 
     def test_checkpoint(self, local_hyperdrive_pool: DeployedHyperdrivePool):
         """Checks that the Hyperdrive checkpoint matches what is returned from the smart contract.
 
         All arguments are fixtures.
         """
-        uri: URI | None = cast(HTTPProvider, local_hyperdrive_pool.web3.provider).endpoint_uri
-        rpc_uri = uri if uri else URI("http://localhost:8545")
-        abi_dir = "./packages/hyperdrive/src/abis"
-        hyperdrive_contract_addresses: HyperdriveAddresses = local_hyperdrive_pool.hyperdrive_contract_addresses
-        eth_config = EthConfig(artifacts_uri="not used", rpc_uri=rpc_uri, abi_dir=abi_dir)
-        hyperdrive = HyperdriveInterface(eth_config, addresses=hyperdrive_contract_addresses)
-        checkpoint_id = hyperdrive.calc_checkpoint_id(block_timestamp=hyperdrive.current_pool_state.block_time)
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        checkpoint_id = hyperdrive_interface.calc_checkpoint_id(
+            block_timestamp=hyperdrive_interface.current_pool_state.block_time
+        )
         # TODO: remove cast when pypechain consolidates dataclasses.
```

Comment:
you can remove this TODO

Reply:
done in 9fee6d551c142a93c069027ca22679f3cd037182